### PR TITLE
Update TestGuide.md

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md [hunk 1]
```diff
@@ -18,7 +18,11 @@ sdk\<service name>\Azure.ResourceManager.<service>\tests\Scenario
 sdk\<service name>\Azure.ResourceManager.<service>\tests\SessionRecords
 ```
 
-**Note**: Considering that in Git directories exist implicitly, so you might need to create the `Scenario` and `SessionRecords` directories by yourself after cloning the repo.
+**Note**: 
+
+1. Considering that in Git directories exist implicitly, so you might need to create the `Scenario` and `SessionRecords` directories by yourself after cloning the repo.
+
+2. The recording files under SessionRecords will eventually be migrated to the assets repository by the test proxy. For more details, please refer to [these steps](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md#test-proxy).
```
Comment (live1206) [inline sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:25]: If all the recording files have been migrated to assets repo already, do we still need to mention migration in this doc? Consider making it a step of the test setup.

File: sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md [hunk 2]
```diff
@@ -6,19 +6,25 @@ We use [NUnit 3][nunit] as our testing framework.
...
 [Azure.Core.TestFramework's testing framework][core_tests] provides a set of reusable primitives that simplify writing tests for new Azure SDK libraries.
 
+The installation of `Test proxy` is a prerequisite for running the tests. Please run the `test-proxy --version` to ensure that test proxy is installed on your computer. If it is not installed, Please refer to [this document](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md#installation) for the installation of the test proxy.
+
 ## Testing project structure
 
 With the help of [Azure.ResourceManager.Template][mgmt_template], the basic testing project and file structure(see following for details) will be generated under `sdk\<service name>\Azure.ResourceManager.<service>\tests` directory.
 
 ```text
+sdk\<service name>\Azure.ResourceManager.<service>\assets.json
```
Comment (live1206) [inline sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:16]: The folder structure has been updated in the template already?

File: sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md [hunk 3]
```diff
@@ -85,6 +91,35 @@ You can also run tests via the command line using `dotnet test`, but that will r
 
 If you are using system environment variables, make sure to restart Visual Studio or the terminal after setting or changing the environment variables.
 
+## Test proxy
+
+Using the test proxy tool, migrate the local recording files to the external repository [azure-sdk-assets](https://github.com/Azure/azure-sdk-assets).
```
Comment (live1206) [inline sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:96]: No migration needed anymore?

File: sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md [hunk 4]
```diff
@@ -6,19 +6,25 @@ We use [NUnit 3][nunit] as our testing framework.
...
 sdk\<service name>\Azure.ResourceManager.<service>\tests\Scenario
-sdk\<service name>\Azure.ResourceManager.<service>\tests\SessionRecords
 ```
 
-**Note**: Considering that in Git directories exist implicitly, so you might need to create the `Scenario` and `SessionRecords` directories by yourself after cloning the repo.
+**Note**: 
+
+1. Considering that in Git directories exist implicitly, so you might need to create the `Scenario` directories by yourself after cloning the repo.
+
+2. There is a reserved field `<service name>` in the `assets.json` file. Please replace it with the correct value before starting to write the tests.
```
Comment (live1206) [inline sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:27]: Could you give an example of the reserved field of `assets.json` here?

File: sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md (hunk 2 above)
Comment (dvbb) [reply sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:16]: yes, on this pr i updated the template

File: sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md (hunk 3 above)
Comment (dvbb) [reply sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:96]: assets.json has been added to the template, so  test framework will save recording files to `/.assets`, only we run `test-proxy push -a ./assets.json` to push local files to remote can pass ci  playback

Comment (live1206) [reply sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md:96]: Yes, I mean we can remove this paragraph, there is no migration needed.
The next paragraph is talking about where the recording files are stored.

**Review by live1206 — APPROVED** [review]

**Pull request merged**